source/manual/errors.html.md

```diff
@@ -1,109 +1,50 @@
-
-## Principles
-
-### 1. When something goes wrong, we should be notified
-
-Applications should report exceptions to Sentry. Applications must not swallow errors.
-
-### 2. Notifications should be actionable
-
-Sentry notifications should be something that requires a developer of the app to do something about it. Not just a piece of information.
-
-### 3. Applications should not error
-
-The goal of GOV.UK is that applications should not error. When something goes wrong it should be fixed.
-
-## Classifying errors
-
-### Bug
-
-A code change makes the application crash.
-
-Desired behaviour: error is sent to Sentry, developers are notified and fix the error. Developers mark the error in Sentry as `Resolved`. This means a recurrence of the error will alert developers again.
-
-### Intermittent errors without user impact
-
-Frontend applications often see timeouts when talking to the content-store.
-
-There's no or little user impact because the request will be answered by the caching layer.
-
-Example: <https://sentry.io/govuk/app-finder-frontend/issues/352985400>
+## How we categorise errors
 
-Desired behaviour: error is not sent to Sentry. Instead, we rely on Smokey and Icinga checks to make sure we the site functions.
-
-### Intermittent errors with user impact
-
-Publishing applications sometimes see timeouts when talking to publishing-api. This results in the publisher seeing an error page and possibly losing data.
-
-Example: <https://sentry.io/govuk/app-content-tagger/issues/367277928>
-
-Desired behaviour: apps handle these errors better, for example by offloading the work to a Sidekiq worker. Since these errors aren't actionable, they should not be reported to Sentry. They should be tracked in Graphite.
-
-### Intermittent retryable errors
-
-Sidekiq worker sends something to the publishing-api, which times out. Sidekiq retries, the next time it works.
-
-Desired behaviour: errors are not reported to Sentry until retries are exhausted. See [this PR for an example](https://github.com/alphagov/content-data-api/pull/353).
-
-Relevant: https://github.com/getsentry/raven-ruby/pull/784
-
-### Expected environment-based errors
-
-MySQL errors on staging while data sync happens.
-
-Example: <https://sentry.io/govuk/app-whitehall/issues/343619055>
-
-Desired behaviour: our environment is set up such that these errors do not occur.
-
-### Bad request errors
-
-User makes a request the application can't handle ([example][bad-request]).
+Sometimes applications will encounter exceptions. This policy describes what we should do for different types of errors. It was first proposed in [RFC 87](https://github.com/alphagov/govuk-rfcs/blob/master/rfc-087-dealing-with-errors.md).
 
-Often happens in [security checks](https://sentry.io/govuk/app-frontend/issues/400074979).
+### High priority errors
```

Comment:
How have we categorised errors into high and low here?

Reply:
The two kinds of error we've got here is based on the previous version of this manual. I've introduced the terminology 'high' and 'low' as a way to refer to them.

- If we're just logging the errors and not actively doing anything about them, it seems fair to say these are 'low priority'.

- If we alert on the errors and are expected to act on that alerts, it seems fair to say these are 'high priority'.

In reality, I don't think people pay as much attention to Sentry as this manual suggests they should, but that's a different problem.

Does that help? I'm open to other terminology, if you have any suggestions.